### ISSUE TITLE
Need access to Rows to set styles, templates, etc

### ISSUE THREAD

**Issue by dapug** [akgulebubekir/Xamarin.Forms.DataGrid#104]
I need to highlight a Row based on a property value in my collection/ItemSource.  Similar to WPF DataGrid.ItemContainerStyle, mentioned here:
https://stackoverflow.com/questions/6957714/how-to-highlight-a-gridview-row-if-some-object-value-is-greater-than-another

I tried setting ActiveRowColor to some color and then programmatically set the SelectedItem, however, I then get the error: `Datagrid must be SelectionEnabled=true to set ActiveRowColor`. This is because I don't actually want user selectable rows, I just want to highlight a row (display only), so of course I have `SelectionEnabled=false`.

Another attempt was to loop through and set properties on the actual rows, but this didn't work either

```C#
    foreach(RowDefinition row in myGrid.RowDefinitions)
                {
                    row.SetValue(Xamarin.Forms.DataGrid.DataGrid.RowsBackgroundColorPaletteProperty, "#8899AA");
                }
```

Basically, this is a feature request, though I am not sure what the simplest way to implement it is (an ItemContainerStyle might be a bit overkill).  Even something as simple as exposing access to the style/colors in the grids Row collection would allow me to set a style on a row programmatically.  Removing the requirement of SelectionEnabled=true would also be a good start (should be able to programmatically "select" an item even if the user can't).

**Comment by akgulebubekir** [comment]
`RowsBackgroundColorPalette ` implements `IColorProvider `that you can use to set all row background one by one with `Color GetColor(int rowIndex, object item)` .

**Comment by akgulebubekir** [comment]
Sample Code : [here](https://github.com/akgulebubekir/Xamarin.Forms.DataGrid/issues/106#issuecomment-462699818)
